gensim/downloader.py

```diff
@@ -0,0 +1,298 @@
+
+
+def get_filename(dataset):
+    """Function of retrieving the filename of corpus/model.
+
+    Args:
+        dataset(string): Name of the corpus/model.
+
+    Returns:
+        string: Returns the filename of the model/corpus.
+    """
+    data = info()
+    corpora = data['gensim']['corpus']
+    models = data['gensim']['model']
+    if dataset in corpora:
+        return data['gensim']['corpus'][dataset]["filename"]
+    elif dataset in models:
+        return data['gensim']['model'][dataset]["filename"]
+
+
+def load(dataset, return_path=False):
+    """Loads the corpus/model to the memory, if return_path is False.
+
+    Args:
+        dataset(string): Name of the corpus/model.
+        return_path(bool): Determines whether to return model/corpus file path.
+
+    Returns:
+        string: Returns the path to the model/corpus, if return_path is True.
+    """
+    file_name = get_filename(dataset)
+    if file_name is None:
+        raise Exception(
+            "Incorrect model/corpus name. Choose the model/corpus from the list "
+            "\n {}".format(json.dumps(info(), indent=4)))
+    folder_dir = os.path.join(base_dir, dataset)
+    file_dir = os.path.join(folder_dir, file_name)
+    if not os.path.exists(folder_dir) or get_data_status(dataset) != "installed":
+        _download(dataset)
+    if return_path:
+        return file_dir
+    else:
+        sys.path.insert(0, base_dir)
+        module = __import__(dataset)
+        data = module.load_data()
+        return data
+
+
+def data_links(dataset):
+    """Function for retrieving the links of the models/corpus which are not stored in github releases
+
+    Args:
+        dataset(string): Name of the corpus/model.
+
+    Returns:
+        string: Returns the link of the model/corpus.
+    """
+    url = "https://raw.githubusercontent.com/chaitaliSaini/Corpus_and_models/master/links.json"
+    response = urlopen(url)
+    data = response.read().decode("utf-8")
+    data = json.loads(data)
+    if dataset in data['data_links']:
+        return data['data_links'][dataset]['link']
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(description="Gensim console API", usage="python -m gensim.api.downloader  [-h] [-d dataset_name | -i dataset_name | -c]")
+    group = parser.add_mutually_exclusive_group()
+    group.add_argument("-d", "--download", metavar="dataset_name", nargs=1, help="To download a corpus/model : python -m gensim -d corpus/model name")
+    group.add_argument("-i", "--info", metavar="dataset_name", nargs=1, help="To get information about a corpus/model : python -m gensim -i model/corpus name")
+    group.add_argument("-c", "--catalogue", help="To get the list of all models/corpus stored : python -m gensim -c", action="store_true")
+    args = parser.parse_args()
+    if args.download is not None:
+        data_path = load(args.download[0], return_path=True)
+        logger.info("Data has been installed and data path is %s", data_path)
```

Comment:
I don't see any logging setup here in `__main__`. Where will this logging go?

We don't want any printing inside a *library*, but from a user-invoked top-level script, printing is fine.